### PR TITLE
Utils improvement

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by divadsn — COMMENTED** [review]
There is no point in parsing a String to a String tbh xD

**Comment by ghost** [comment]
i'm drunk, lemme undo dat
warnings trigger me

**Comment by skittles9823** [comment]
@alamo18 Lmao nice one Morgan.

**Comment by ghost** [comment]
Is this ok?

**Comment by bohrabhijeet** [comment]
Nice one! Morgan rocks!

File: app/src/main/java/ch/deletescape/lawnchair/Utilities.java [hunk 1]
```diff
@@ -1113,19 +1108,11 @@ public static boolean checkOutdatedLawnfeed(Context context) {
...
-
-        // Create a pending intent so the application is restarted after System.exit(0) was called.
-        // We use an AlarmManager to call this intent in 100ms
-        PendingIntent mPendingIntent = PendingIntent.getActivity(context, 0, startActivity, PendingIntent.FLAG_CANCEL_CURRENT);
-        AlarmManager mgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
-        mgr.set(AlarmManager.RTC, System.currentTimeMillis() + 100, mPendingIntent);
+        Intent startActivity = new Intent(context, Launcher.class);
+        startActivity.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        context.startActivity(startActivity);
+        ((Activity) context).finish();
```
Comment (suphon-t) [inline app/src/main/java/ch/deletescape/lawnchair/Utilities.java]: Finishing and restarting an activity is not equal to restarting the launcher. There are some static references that will still persist when all activities are destroyed.

Comment (suphon-t) [reply app/src/main/java/ch/deletescape/lawnchair/Utilities.java]: Also, you can't just cast a Context to an Activity and expect it to always work. If you really need to, change the method signature or add a check before casting.

**Comment by ghost** [comment]
Summary of my PR:
1. Gradle 3.1.0-alpha08 => 3.2.0-alpha03
2. restartLauncher improvement (start Launcher.class, clear top then finish. There is no need to import PackageManager and AlarmManager)
3. Cleanup

**Comment by bohrabhijeet** [comment]
@alamo18 Great work!

File: app/src/main/java/ch/deletescape/lawnchair/Utilities.java (hunk 1 above)
Comment (ghost) [reply app/src/main/java/ch/deletescape/lawnchair/Utilities.java]: ok will do after class

**Pull request merged**